dags/analyzer_helper/tests/integration/test_extract_raw_members.py

```diff
@@ -0,0 +1,111 @@
+
+class TestExtractRawMembers(unittest.TestCase):
+
+    def setUp(cls):
+        cls.neo4jConnection = Neo4jConnection()
+        cls.driver = cls.neo4jConnection.connect_neo4j()
+        cls.test_forum_endpoint = "https://test-forum.discourse.org"
+        cls.platform_id = "test_platform"
+        cls.extractor = ExtractRawMembers(cls.test_forum_endpoint, cls.platform_id)
+        cls.rawmembers_collection = cls.extractor.rawmembers_collection
+        
+        cls.rawmembers_collection.insert_many([
+            {'id': 1, 'joined_at': datetime(2023, 7, 1), 'avatar': 'avatar1.png', 'badgeIds': [1, 2]},
+            {'id': 2, 'joined_at': datetime(2023, 2, 2), 'avatar': 'avatar2.png', 'badgeIds': [3]}
+        ])
+
+        with cls.driver.session() as session:
+
+            result_forum = session.run(
+                """
+                CREATE (f:DiscourseForum {endpoint: $forum_endpoint})
+                RETURN id(f) AS id
+                """,
+                forum_endpoint=cls.test_forum_endpoint
+            )
+            cls.forum_id = result_forum.single()["id"]
+            
+            # Create user1 and relate to forum
+            result1 = session.run(
+                """
+                MATCH (f:DiscourseForum {endpoint: $forum_endpoint})
+                CREATE (u:DiscourseUser {id: 'user1', avatarTemplate: 'avatar1', createdAt: '2023-07-01'})
+                -[:HAS_JOINED]->(f)
+                CREATE (u)-[:HAS_BADGE]->(:Badge {id: 'badge1'})
+                RETURN id(u) AS id
+                """,
+                forum_endpoint=cls.test_forum_endpoint
+            )
+            cls.user1_id = result1.single()["id"]
+
+            # Create user2 and relate to forum
+            result2 = session.run(
+                """
+                MATCH (f:DiscourseForum {endpoint: $forum_endpoint})
+                CREATE (u:DiscourseUser {id: 'user2', avatarTemplate: 'avatar2', createdAt: '2023-07-02'})
+                -[:HAS_JOINED]->(f)
+                CREATE (u)-[:HAS_BADGE]->(:Badge {id: 'badge2'})
+                RETURN id(u) AS id
+                """,
+                forum_endpoint=cls.test_forum_endpoint
+            )
+            cls.user2_id = result2.single()["id"]
+
+    def tearDown(cls):
+        with cls.driver.session() as session:
+            session.run("MATCH (n) DETACH DELETE n")
+        cls.extractor.close()
+        cls.driver.close()
+        cls.rawmembers_collection.delete_many({})
+
+    def test_fetch_member_details(self):
+        member_details = self.extractor.fetch_member_details()
+        
+        expected_members = [
+            {
+                'id': 'user1',
+                'avatar': 'avatar1',
+                'joined_at': '2023-07-01',
+                'badgeIds': ['badge1'],
+            },
+            {
+                'id': 'user2',
+                'avatar': 'avatar2',
+                'joined_at': '2023-07-02',
+                'badgeIds': ['badge2']
```

Comment:
This structure is not what we expect. It should follow the structure we defined before https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=rawmembers%20collection%20(transformed%20guildmembers)

You could skip the `left_at` field, but others should be there as defined.